crates/dbsp/src/trace/ord/file/val_batch.rs

```diff
@@ -540,21 +552,24 @@ where
                                     while let Some(value1) =
                                         read_filtered(cursor1, value_filter, tmp_v1)
                                     {
-                                        self.copy_value(output, cursor1, value1);
-                                        n += 1;
+                                        n += self.copy_value(output, cursor1, value1, map_func);
+                                        cursor1.move_next().unwrap();
                                     }
                                     return;
                                 };
                                 match value1.cmp(value2) {
-                                    Ordering::Less => self.copy_value(output, cursor1, value1),
+                                    Ordering::Less => {
+                                        n += self.copy_value(output, cursor1, value1, map_func);
+                                        cursor1.move_next().unwrap();
+                                    }
                                     Ordering::Equal => {
                                         let td1 = unsafe { cursor1.aux(td1) }.unwrap();
                                         let td2 = unsafe { cursor2.aux(td2) }.unwrap();
                                         if let Some(map_func) = &map_func {
                                             merge_map_times(td1, td2, map_func, td);
                                         } else {
-                                            td1.sort_unstable();
-                                            td2.sort_unstable();
+                                            //debug_assert!(td1.is_sorted());
```

Comment:
are these true but too expensive, or should this be deleted?

Reply:
Our vectors don't have the `is_sorted` method yet.